test/test_fcntl.rs

```diff
@@ -232,3 +232,46 @@ mod test_posix_fadvise {
         assert_eq!(errno, Errno::ESPIPE as i32);
     }
 }
+
+#[cfg(any(target_os = "linux",
+          target_os = "android",
+          target_os = "emscripten",
+          target_os = "fuchsia",
+          any(target_os = "wasi", target_env = "wasi"),
+          target_env = "freebsd"))]
+mod test_posix_fallocate {
+
+    use tempfile::NamedTempFile;
+    use std::{io::Read, os::unix::io::{RawFd, AsRawFd}};
+    use nix::errno::Errno;
+    use nix::fcntl::*;
+    use nix::unistd::pipe;
+
+    #[test]
+    fn test_success() {
+        const LEN: usize = 100;
+        let mut tmp = NamedTempFile::new().unwrap();
+        let fd = tmp.as_raw_fd();
+        let res = posix_fallocate(fd, 0, LEN as libc::off_t).unwrap();
```

Comment:
This test will fail if /tmp resides on a file system that doesn't support `posix_fallocate`, like ZFS.  The test must be adjusted to handle that error.  Unfortunately, POSIX stipulates that the errno in that case shall be `EINVAL`, which can also indicate several other problems.

Reply:
Thanks! I wasn't aware of this case.

Reply:
I have added in some explanation in the test, in case the test fails possibly due to filesystem not supporting this operation.